src/services/resource.js

```diff
@@ -210,6 +327,394 @@ module.exports = class resourceHelper {
+				user_id: {
+					[Op.notIn]: [user_id],
+				},
+			}
+			if (searchText != '')
+				resourceFilter.title = {
+					[Op.iLike]: '%' + searchText + '%',
+				}
+
+			if (common.TYPE in queryParams && queryParams[common.TYPE]) {
+				resourceFilter.type = queryParams[common.TYPE].split(',')
+			}
+
+			const response = await resourceQueries.resourceList(
+				resourceFilter,
+				[
+					'id',
+					'title',
+					'type',
+					'organization_id',
+					'status',
+					'user_id',
+					'submitted_on',
+					'last_reviewed_on',
+					'created_at',
+					'meta',
+				],
+				sort,
+				page,
+				limit
+			)
+
+			if (response.length === 0) {
+				return responses.successResponse({
+					statusCode: httpStatusCode.ok,
+					message: 'RESOURCE_LISTED_SUCCESSFULLY',
+					result: [],
+				})
+			}
+
+			const uniqueCreatorIds = utils.getUniqueElements(
+				response.result.map((item) => {
+					return item.user_id
+				})
+			)
+
+			const userDetails = await this.fetchUserDetails(uniqueCreatorIds)
+			const orgDetails = await this.fetchOrganizationDetails(uniqueOrganizationIds)
+
+			result.data = response.result.map((item) => {
+				let returnValue = item
+
+				if (item.meta?.notes) returnValue.notes = item.meta.notes
+				if (inProgressResources.includes(item.id)) {
+					returnValue.reviewer_status = common.REVIEW_STATUS_INPROGRESS
+				}
+				returnValue.creator =
+					userDetails[item.user_id] && userDetails[item.user_id].name ? userDetails[item.user_id].name : ''
+				returnValue.organization = orgDetails[item.organization_id]
+				delete item.user_id
+				delete item.organization_id
+				delete returnValue.meta
+				return returnValue
+			})
+			result.count = response.count
+			result.in_progress_count = in_progress_count
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'RESOURCE_LISTED_SUCCESSFULLY',
+				result,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async findSequentialResources(organization_id, roles, resourceTypes = []) {
+		// get unique user roles
+		const userRoleTitles = utils.getUniqueElements(roles.map((item) => item.title))
+
+		// fetch the resource wise review levels
+		const resourceWiseLevels = await this.fetchResourceReviewLevel(organization_id, userRoleTitles, resourceTypes)
+		let resourceTypeStagesConfig = []
+		resourceTypes.filter((type) => {
+			if (resourceWiseLevels[type]) {
+				resourceTypeStagesConfig.push({
+					[Op.and]: [{ type: type }, { next_stage: resourceWiseLevels[type] }],
+				})
+			}
+		})
+		const resourceFilter = {
+			[Op.or]: resourceTypeStagesConfig,
+			organization_id,
+			status: { [Op.in]: [common.RESOURCE_STATUS_SUBMITTED, common.RESOURCE_STATUS_IN_REVIEW] },
+		}
+		const resourcesDetails = await resourceQueries.findAll(resourceFilter, ['id'])
+		if (resourcesDetails) {
+			const resoureId = resourcesDetails.map((item) => {
+				return item.id
+			})
+			return resoureId
+		}
+		return []
```

Comment:
@adithyadinesh0412  assign an empty array result in top and push the resource ids into that array. Then return result